src/ray/common/ray_config_def.h

```diff
@@ -661,16 +661,22 @@ RAY_CONFIG(bool, isolate_workers_across_task_types, true)
 /// ServerCall instance number of each RPC service handler
 RAY_CONFIG(int64_t, gcs_max_active_rpcs_per_handler, 100)
 
-/// grpc keepalive sent interval
+/// grpc keepalive sent interval for server.
 /// This is only configured in GCS server now.
-/// NOTE: It is not ideal for other components because
+RAY_CONFIG(int64_t, grpc_keepalive_time_ms, 10000)
+
+/// grpc keepalive timeout.
+RAY_CONFIG(int64_t, grpc_keepalive_timeout_ms, 20000)
```

Comment:
I wonder when a ping is sent, does it reset the timeout? If so, this will never have timeout right? E.g. the server will keep sending every 10 seconds, and never timeout since it the timer gets reset every 10sec.

Reply:
This config has fixed bugs before, so I believe it is not reset

Reply:
I see - so the timeout is only reset when *receive* a keepalive response.